crates/bevy_ecs/macros/src/lib.rs

```diff
@@ -423,6 +424,411 @@ pub fn derive_system_param(input: TokenStream) -> TokenStream {
     })
 }
 
+static READONLY_ATTRIBUTE_NAME: &str = "readonly";
+static FILTER_ATTRIBUTE_NAME: &str = "filter";
+
+/// Implement `WorldQuery` to use a struct as a parameter in a query
+#[proc_macro_derive(Fetch, attributes(readonly, filter))]
+pub fn derive_fetch(input: TokenStream) -> TokenStream {
+    let ast = parse_macro_input!(input as DeriveInput);
+
+    let FetchImplTokens {
+        struct_name,
+        fetch_struct_name,
+        state_struct_name,
+        fetch_trait_punctuated_lifetimes,
+        impl_generics,
+        ty_generics,
```

Comment:
Avoid abbreviations here.

Reply:
I'm open to fixing it, but this is a pretty common abbreviation for derive macros. Also, we already use it in `derive_label`, for example.

Reply:
Ah okay; we can leave it be just fine if it's a convention.